extensions/parquet/base/src/main/java/io/deephaven/parquet/base/RowGroupReaderImpl.java

```diff
@@ -76,15 +80,16 @@ public ColumnChunkReaderImpl getColumnChunk(@NotNull final List<String> path) {
 
         OffsetIndex offsetIndex = null;
         if (columnChunk.isSetOffset_index_offset()) {
-            try (final SeekableByteChannel readChannel = channelsProvider.getReadChannel(rootPath)) {
+            try (final SeekableChannelContext channelContext = channelsProvider.makeContext();
```

Comment:
Any time you have this pattern, you might want to consider whether there's a higher level from which to pass the context. In this case, `getColumnLocation` is looping over multiple calls to `getColumnChunk`.

Reply:
Nice catch.
I am a bit confused about implementing this one though. 
So `RegionedColumnSourceManager` calls `getColumnLocation` for each column which internally calls `getColumnChunk` for each row group. I was thinking of using a single ChannelContext for all of these, but then its not very clear to me where I will close this ChannelContext object.
Similar problem is happening for me with closing `S3SeekableChannelProvider`. I can't figure out where to call its close from.
Can you please have a look at my latest commit and suggest?